### PR TITLE
Human Task Editor for Eclipse Using Web UI Platform

### PR DESCRIPTION
**Pull request by Drifftr**

### CODE REVIEW

**Comment by hasithaa** [comment]
I am getting following build error. 

[INFO] WSO2 BPS Tools .................................... SUCCESS [0.143s]
[INFO] WSO2 BPS Tooling Plugins .......................... SUCCESS [0.005s]
[INFO] Developer Studio BPEL Help ........................ SUCCESS [2.010s]
[INFO] Developer Studio BPEL Common Model ................ SUCCESS [1.548s]
[INFO] Developer Studio BPEL Model ....................... SUCCESS [2.459s]
[INFO] Developer Studio BPEL Deploy Model ................ SUCCESS [0.615s]
[INFO] Developer Studio BPEL Common UI ................... SUCCESS [0.744s]
[INFO] Developer Studio BPEL Common WSDL ................. SUCCESS [0.272s]
[INFO] Developer Studio BPEL Runtimes .................... SUCCESS [0.480s]
[INFO] Developer Studio BPEL WSIL Model .................. SUCCESS [0.266s]
[INFO] Developer Studio BPEL XPath ....................... SUCCESS [0.432s]
[INFO] Developer Studio BPEL Validator ................... SUCCESS [0.835s]
[INFO] Developer Studio BPEL UI .......................... SUCCESS [2.982s]
[INFO] Developer Studio BPEL Deploy UI ................... SUCCESS [0.606s]
[INFO] Developer Studio BPEL Runtime ..................... SUCCESS [0.636s]
[INFO] Developer Studio BPEL Cheatsheet .................. SUCCESS [0.227s]
[INFO] Developer Studio Humantask Model .................. SUCCESS [0.537s]
[INFO] Developer Studio Humantask UI ..................... SUCCESS [0.594s]
[INFO] Developer Studio BPEL Export ...................... SUCCESS [0.415s]
[INFO] WSO2 Developer Studio BPEL Artifact ............... SUCCESS [0.878s]
[INFO] Developer Studio BPEL Core ........................ SUCCESS [0.189s]
[INFO] Developer Studio BPEL Connector ................... SUCCESS [0.335s]
[INFO] Developer Studio Humantask Editor ................. FAILURE [0.341s]
[INFO] WSO2 BPS Tooling Feature .......................... SKIPPED
[INFO] WSO2 Developer Studio BPS Feature ................. SKIPPED
[INFO] WSO2 Developer Studio Human Task Editor ........... SKIPPED
[INFO] WSO2 Developer Studio - BPS P2 Repository Parent .. SKIPPED
[INFO] WSO2 Developer Studio Kernel - BPS Main P2 Repository  SKIPPED
[INFO] WSO2 Developer Studio - BPS Composite P2 Repository  SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2:33.690s
[INFO] Finished at: Mon Apr 18 17:06:08 IST 2016
[INFO] Final Memory: 247M/1063M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:0.23.0:compile (default-compile) on project org.wso2.developerstudio.humantaskeditor: Compilation failure: Compilation failure:
[ERROR] /home/hasi/git/wso2/devstudio-tooling-bps/plugins/org.wso2.developerstudio.humantaskeditor/src/org/wso2/developerstudio/humantaskeditor/editors/HumanTaskMultiPageEditor.java:[77]
[ERROR] humanTaskUIEditor.setEditorFunctionExecutor(new EditorContentFunction());
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The method setEditorFunctionExecutor(EditorContentFunction) is undefined for the type HumanTaskUIEditor
[ERROR] /home/hasi/git/wso2/devstudio-tooling-bps/plugins/org.wso2.developerstudio.humantaskeditor/src/org/wso2/developerstudio/humantaskeditor/editors/EditorContentFunction.java:[30]
[ERROR] import org.wso2.developerstudio.eclipse.webui.core.editor.AbstractEditorFunctionExecutor;
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The import org.wso2.developerstudio.eclipse.webui.core.editor.AbstractEditorFunctionExecutor cannot be resolved
[ERROR] /home/hasi/git/wso2/devstudio-tooling-bps/plugins/org.wso2.developerstudio.humantaskeditor/src/org/wso2/developerstudio/humantaskeditor/editors/EditorContentFunction.java:[34]
[ERROR] public class EditorContentFunction implements AbstractEditorFunctionExecutor {
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] AbstractEditorFunctionExecutor cannot be resolved to a type
[ERROR] /home/hasi/git/wso2/devstudio-tooling-bps/plugins/org.wso2.developerstudio.humantaskeditor/src/org/wso2/developerstudio/humantaskeditor/editors/EditorContentFunction.java:[41]
[ERROR] public Object executeFunction(String functionName, Object[] parameters) {
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] The method executeFunction(String, Object[]) of type EditorContentFunction must override or implement a supertype method
[ERROR] 4 problems (4 errors)
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :org.wso2.developerstudio.humantaskeditor

File: feature-updater/pom.xml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    <name>WSO2 Developer Studio BPS Tooling Updates Publisher</name>
+   
+ 
+    <repositories>
+    </repositories>
+    <pluginRepositories>
+       
+    </pluginRepositories>
+
+<build>
```
Comment (hasithaa) [inline feature-updater/pom.xml]: Fix styling issues.

File: plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/hteditor.js [hunk 2]
```diff
@@ -0,0 +1,1053 @@
...
+function createFile(currentTaskName,state) { //createFile
+    // Create Task WSDL
+    var serviceURL = $('#' + currentTaskName + 'wrapper #taskCallbackServiceURL').val();
+    var operationName = $('#' + currentTaskName + 'wrapper #taskCallbackOperationName')
+        .val();
+    var serviceName = $('#' + currentTaskName + 'wrapper #taskCallbackServiceName').val();
+
+    var outputElements = $('#' + currentTaskName + 'wrapper #outputmappingTable tr');
+ $.get('resources/dummy.wsdl', function(data) {
+
```
Comment (hasithaa) [inline plugins/org.wso2.developerstudio.humantaskeditor/HumanTaskEditor/hteditor.js]: Formatting issues. Remove unwanted white spaces.

File: feature-updater/publish.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
+ #!/bin/bash
+echo "\nCreating 'Feature Update' for BPS Tooling Feature\n"
```
Comment (hasithaa) [inline feature-updater/publish.sh]: Missing  Licences header.

File: plugins/org.eclipse.bpel.ui.noEmbeddedEditors/src/org/eclipse/bpel/ui/Messages.java [hunk 4]
```diff
@@ -78,6 +78,10 @@ private Messages() {
 	public static String NamespaceMappingDialog_9;
 	public static String OperationAdapter_0;
 	public static String OutputMessageAdapter_0;
+	public static String PaletteFactory_Actions;
```
Comment (hasithaa) [inline plugins/org.eclipse.bpel.ui.noEmbeddedEditors/src/org/eclipse/bpel/ui/Messages.java]: Fix formatting issues.

File: plugins/org.eclipse.bpel.ui.noEmbeddedEditors/src/org/eclipse/bpel/ui/properties/AbstractStyledTextExpressionSection.java [hunk 5]
```diff
@@ -34,256 +34,252 @@
...
  * Base class with some shared behavior for details panes that edit a through a {@link StyledText}.
+ * 
  * @author Michal Chmielewski (michal.chmielewski@oracle.com)
- * @author Vincent Zurczak - EBM WebSourcing (Merge this class with TextSection, simplify it by only using the XPath viewer)
+ * @author Vincent Zurczak - EBM WebSourcing (Merge this class with TextSection, simplify it by only using the XPath
+ *         viewer)
  * @author Lorenzo Bettini abstract class for all StyledText based expression sections
- * FIXME: There are probably cases that are not handled correctly. To be fixed on the fly.
- * FIXME: Undo is not supported correctly.
+ *         FIXME: There are probably cases that are not handled correctly. To be fixed on the fly.
```
Comment (hasithaa) [inline plugins/org.eclipse.bpel.ui.noEmbeddedEditors/src/org/eclipse/bpel/ui/properties/AbstractStyledTextExpressionSection.java]: Fix Formatting issues. Replace tab with spaces.

File: feature-updater/publish.sh (hunk 3 above)
Comment (Drifftr) [inline feature-updater/publish.sh]: Fixed in https://github.com/wso2/devstudio-tooling-bps/commit/3f440647b4d817de41875b600c296961b2662bc1

File: feature-updater/pom.xml (hunk 1 above)
Comment (Drifftr) [inline feature-updater/pom.xml]: Fixed in https://github.com/wso2/devstudio-tooling-bps/commit/3f440647b4d817de41875b600c296961b2662bc1

File: plugins/org.eclipse.bpel.ui.noEmbeddedEditors/src/org/eclipse/bpel/ui/Messages.java (hunk 4 above)
Comment (Drifftr) [inline plugins/org.eclipse.bpel.ui.noEmbeddedEditors/src/org/eclipse/bpel/ui/Messages.java]: Fixed in 3f44064

File: plugins/org.eclipse.bpel.ui.noEmbeddedEditors/src/org/eclipse/bpel/ui/properties/AbstractStyledTextExpressionSection.java (hunk 5 above)
Comment (Drifftr) [inline plugins/org.eclipse.bpel.ui.noEmbeddedEditors/src/org/eclipse/bpel/ui/properties/AbstractStyledTextExpressionSection.java]: Fixed in 3f44064

**Pull request merged**